tests/optimize/test_spsa.py

```diff
@@ -0,0 +1,337 @@
+# Copyright 2018-2022 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Tests for the SPSA optimizer"""
+import pytest
+
+import pennylane as qml
+from pennylane import numpy as np
+
+
+univariate = [(np.sin), (lambda x: np.exp(x / 10.0)), (lambda x: x**2)]
+
+multivariate = [
+    (lambda x: np.sin(x[0]) + np.cos(x[1])),
+    (lambda x: np.exp(x[0] / 3) * np.tanh(x[1])),
+    (lambda x: np.sum([x_**2 for x_ in x])),
+]
+
+
+class TestSPSAOptimizer:
+    """Test the SPSA optimizer"""
```

Comment:
It would be good to first test that the various attributes are set correctly on the `optimizer` object.